tests/component-tests/textfieldSlider/js/TextfieldSliderTests.js

```diff
@@ -21,76 +21,160 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
 
         jqUnit.module("TextfieldSlider Tests");
 
-        fluid.tests.textfieldSlider.createTextfieldSlider = function (options) {
-            return fluid.textfieldSlider(".fl-textfield-slider", options);
+        fluid.tests.textfieldSlider.createTextfieldSliderNative = function (options) {
+            // Make sure we're using the native widget
+            fluid.contextAware.makeChecks({
+                "fluid.prefsWidgetType": {
+                    value: "nativeHTML"
+                }
+            });
+            return fluid.textfieldSlider(".fl-textfield-slider-native", options);
+        };
+
+        fluid.tests.textfieldSlider.createTextfieldSliderJQuery = function (options) {
+            // Override default native widget to use jQuery slider instead
+            fluid.contextAware.makeChecks({
+                "fluid.prefsWidgetType": {
+                    value: "jQueryUI"
+                }
+            });
+
+            return fluid.textfieldSlider(".fl-textfield-slider-jQuery", options);
+        };
+
+        // Returns the value of either a native input[type=range] or a jQuery slider
+        fluid.tests.textfieldSlider.getSliderValue = function (slider) {
+
+            var sliderValue;
+            try {
+                sliderValue = slider.slider("value");
+            } catch (e) {
+                sliderValue = slider.val();
+            }
```

Comment:
The overall strategy isn't right. We should no more have implementation switches based on conditionals in code in our tests than we should in the real implementation. This should be parameterised inside a grade with an invoker whose implementation comes from a grade, just as in the component impl itself.